packages/vite/src/node/server/index.ts

```diff
@@ -500,8 +500,13 @@ export async function createServer(
   middlewares.use(serveRawFsMiddleware(server))
   middlewares.use(serveStaticMiddleware(root, config))
 
-  // spa fallback
+  // run post config hooks
+  // This is applied before the html middleware so that user middleware can
+  // serve custom content instead of index.html.
+  postHooks.forEach((fn) => fn && fn())
```

Comment:
Even shorter 😜

```suggestion
  postHooks.forEach((fn) => fn?.())
```

Reply:
I had to revert this change because it didn't work. It's also harder to read in my opinion and unrelated to the PR, so I think we can probably just leave as is

Reply:
```node
This expression is not callable.
  Not all constituents of type 'void | (() => void)' are callable.
    Type 'void' has no call signatures.

529   postHooks.forEach((fn) => fn?.())
                                ~~
```

Interesting :thinking: 
I'm question my self if this is a TypeScript bug :thinking: